spec.bs

```diff
@@ -2358,38 +2363,140 @@ a {{ReportingBrowserSignals}} |browserSignals|, and a [=direct from seller signa
+          [=generated bid/top level seller debug win report url=], |seller|, and
+          |auctionReportInfo|'s [=auction report info/debug win report urls=].
     1. Otherwise:
-      1. If |generatedBid|'s [=generated bid/bidder debug loss report url=] is not null,
-        then [=list/append=] it to |auctionReportInfo|'s [=auction report info/debug loss report urls=].
-      1. If |generatedBid|'s [=generated bid/seller debug loss report url=] is not null,
-        then [=list/append=] it to |auctionReportInfo|'s [=auction report info/debug loss report urls=].
-      1. If |generatedBid|'s [=generated bid/top level seller debug loss report url=] is
-        not null, then [=list/append=] it to |auctionReportInfo|'s
+      1. [=Collect a single forDebuggingOnly report=] with |generatedBid|'s
+        [=generated bid/bidder debug loss report url=], |generatedBid|'s
+        [=generated bid/interest group=]'s [=interest group/owner=], and |auctionReportInfo|'s
         [=auction report info/debug loss report urls=].
+      1. If |generatedBid|'s [=generated bid/component seller=] is null:
+        1. [=Collect a single forDebuggingOnly report=] with |generatedBid|'s
+          [=generated bid/seller debug loss report url=], |seller|, and
+          |auctionReportInfo|'s [=auction report info/debug loss report urls=].
+      1. Otherwise:
+        1. [=Collect a single forDebuggingOnly report=] with |generatedBid|'s
+          [=generated bid/seller debug loss report url=], |generatedBid|'s
+          [=generated bid/component seller=], |auctionReportInfo|'s
+          [=auction report info/debug loss report urls=].
+        1. [=Collect a single forDebuggingOnly report=] with |generatedBid|'s
+          [=generated bid/top level seller debug loss report url=], |seller|, and
+          |auctionReportInfo|'s [=auction report info/debug loss report urls=].
   1. Return |auctionReportInfo|.
 </div>
 
+### Downsampling ### {#downsampling-header}
+
+*This first introductory paragraph is non-normative.*
+
+Implementations may define their own values for the below constants based on their privacy goal,
+however we supply the below values as a starting point, inspired by what the initial implementation
+of this specification uses:
+  * <dfn>sampling rate</dfn> is 1/1000, which means only sending reports 1/1000 times the
+    {{InterestGroupBiddingAndScoringScriptRunnerGlobalScope/forDebuggingOnly}} API is called.
+  * <dfn>lockout period</dfn> is 3 years.
+  * <dfn>long cooldown rate</dfn> is 1/10, which means 10% of the time putting an adtech in a long
+    cooldown period if the adtech calls the API.
+  * <dfn>long cooldown period</dfn> is 1 year.
+  * <dfn>short cooldown period</dfn> is 2 weeks.
+
+Note: More details of how these numbers are determined can be found in comments here:
+(<a href="https://github.com/WICG/turtledove/issues/632">WICG/turtledove#667</a>).
+
+The design of downsampling [=forDebuggingOnly reports=] has two main goals:

```

Comment:
Could you please update this to having three goals, as in the corresponding section of the explainer?

Reply:
good catch, thanks for noticing!